beaker/precompile.py

```diff
@@ -108,7 +125,16 @@ def assemble(self, client: AlgodClient):
             # +1 to acount for the pushbytes/pushint op
             tv.pc = self._map.get_pcs_for_line(tv.line)[0] + 1
 
-    def hash(self) -> Expr:
+        self.program_pages = [
+            ProgramPage(
+                index=i,
+                _binary=self._binary[i : i + 2048],
+                _hash_digest=hashlib.sha256(self._binary[i : i + 2048]).digest(),
+            )
+            for i in range(0, len(self._binary), 2048)
+        ]
+
+    def hash(self, page_idx: Optional[int] = -1) -> Expr:
```

Comment:
Would also need to tweak the below if case
```suggestion
    def hash(self, page_idx: int = 0) -> Expr:
```